### PR TITLE
openmpi: libfabric support, ucx cleanup

### PR DESCRIPTION
**Pull request by baberlevi**
This separates the changes for openmpi from #6978 and integrates with latest changes from develop.

### CODE REVIEW

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 1]
```diff
@@ -178,11 +179,13 @@ class Openmpi(AutotoolsPackage):
     patch('configure.patch', when="@1.10.1")
     patch('fix_multidef_pmi_class.patch', when="@2.0.0:2.0.1")
 
+    fabrics = ('psm', 'psm2', 'verbs', 'mxm', 'ucx', 'libfabric')
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openmpi/package.py]: adds support for libfabric and ucx and makes fabrics a variable that can be used in when decorators below

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 2]
```diff
@@ -200,7 +203,6 @@ class Openmpi(AutotoolsPackage):
     variant('thread_multiple', default=False,
             description='Enable MPI_THREAD_MULTIPLE support')
     variant('cuda', default=False, description='Enable CUDA support')
-    variant('ucx', default=False, description='Enable UCX support')
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openmpi/package.py]: is a fabric type, moved there

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 3]
```diff
@@ -219,6 +221,9 @@ class Openmpi(AutotoolsPackage):
     provides('mpi@:3.0', when='@1.7.5:')
     provides('mpi@:3.1', when='@2.0.0:')
 
+    if sys.platform != 'darwin':
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openmpi/package.py]: numactl not supported on Mac, but is required. Has been pulling system version in

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 4]
```diff
@@ -229,9 +234,10 @@ class Openmpi(AutotoolsPackage):
     depends_on('hwloc +cuda', when='+cuda')
     depends_on('java', when='+java')
     depends_on('sqlite', when='+sqlite3@:1.11')
-    depends_on('ucx', when='+ucx')
     depends_on('zlib', when='@3.0.0:')
     depends_on('valgrind~mpi', when='+memchecker')
+    depends_on('ucx', when='fabrics=ucx')
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openmpi/package.py]: dependency when clause changed to reflect ucx moving to fabrics list

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 5]
```diff
@@ -229,9 +234,10 @@ class Openmpi(AutotoolsPackage):
...
+    depends_on('libfabric', when='fabrics=libfabric')
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openmpi/package.py]: new dependency for libfabric

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 6]
```diff
@@ -415,12 +423,6 @@ def configure_args(self):
             else:
                 config_args.append('--without-cuda')
 
-        # UCX support
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openmpi/package.py]: now handled by `with_or_without('fabrics')`

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 7]
```diff
@@ -178,11 +179,13 @@ class Openmpi(AutotoolsPackage):
...
     patch('fix_multidef_pmi_class.patch', when="@2.0.0:2.0.1")
 
+    fabrics = ('psm', 'psm2', 'verbs', 'mxm', 'ucx', 'libfabric')
+
     variant(
         'fabrics',
         default=None if _verbs_dir() is None else 'verbs',
-        description='List of fabrics that are enabled',
-        values=('psm', 'psm2', 'verbs', 'mxm'),
+        description=("List of fabrics that are enabled"),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmpi/package.py]: Why the parentheses?

Comment (baberlevi) [reply var/spack/repos/builtin/packages/openmpi/package.py]: I really don't remember why I did that. Removed

**Pull request merged**